src/lib_ccx/hardsubx_classifier.c

```diff
@@ -48,9 +48,10 @@ char *get_ocr_text_wordwise(struct lib_hardsubx_ctx *ctx, PIX *image)
 	{
 		do
 		{
-			char* word = strdup(TessResultIteratorGetUTF8Text(it, level));
+			char* word = TessResultIteratorGetUTF8Text(it, level);
 			if(word==NULL || strlen(word)==0)
 				continue;
+			word = strdup(word);
```

Comment:
This doesn't seem correct to me (but the previous one isn't either).

word = strdup(word);

We're losing the original pointer - assuming TessResultIteratorGetUTF8Text() allocates something (I assume it does, but I don't know for sure) we now have a leak.

Using strdup() seems OK to me IF we're going to be messing with the data later, but in this case we should immediately free the tesseract resource after making a copy, since we no longer need it.

Check this out:
http://tess4j.sourceforge.net/docs/docs-1.2/net/sourceforge/tess4j/TessAPI.html#TessBaseAPIGetIterator(net.sourceforge.tess4j.TessAPI.TessBaseAPI)


    TessBaseAPIGetIterator

    TessAPI.TessResultIterator TessBaseAPIGetIterator(TessAPI.TessBaseAPI handle)

    Get a reading-order iterator to the results of LayoutAnalysis and/or Recognize. ****. WARNING! This class points to data held within the TessBaseAPI class, and therefore can only be used while the TessBaseAPI class still exists and has not been subjected to a call of Init, SetImage, Recognize, Clear, End, DetectOS, or anything else that changes the internal PAGE_RES.

Reply:
Oh yes, I didn't realize that I was creating a memory leak. So something like:
```
char *word_ = TessResultIteratorGetUTF8Text(it, level);
if(word_ == NULL || !strlen(word_))
    continue;
char *word = strdup(word_);
free(word_);
```
This should work fine right?

Reply:
No... you can't use free() to unallocate resources returned by TessResultIteratorGetUTF8Text() - take a look at the link I sent. Most likely (but check - I just browsed the API) it's TessResultIteratorDelete.